doc/maintenance.md

```diff
@@ -246,13 +246,21 @@ publishing the actual release notes at https://www.pygmt.org/latest/changes.html
    the updated release notes, so that other people can help to review and
    collaborate on the changelog curation process described next.
 4. Edit the change list to remove any trivial changes (updates to the README,
-   typo fixes, CI configuration, etc).
+   typo fixes, CI configuration, test updates due to GMT releases, etc).
+5. Sort the items within each section (i.e., New Features, Enhancements, etc.)
+   such that similar items are located near each other (e.g., new wrapped
+   modules, gallery examples, API docs changes) and entries within each group
+   are alphabetical.
+6. Move a few important items from the main sections to the highlights section.
 5. Edit the list of people who contributed to the release, linking to their
```

Comment:
See https://github.com/GenericMappingTools/pygmt/pull/1341#discussion_r654867365:
> > > > How do I know whether any of the current order reflects a case-by-case basis or whether I should update based on `git shortlog -sne`?
> > > 
> > > See previous discussion at [#1047 (comment)](https://github.com/GenericMappingTools/pygmt/pull/1047#discussion_r593880470). Will leave it up to you to take a first stab at the order :) Just to clarify, that line refers to the Zenodo citation order. The release note authors order should go strictly by `git shortlog` (and remove the bots).
> > 
> > OK, I guess I wasn't paying good attention for that release :) Thanks for the note.

> Just on the author list, this is the order from `git shortlog -sne` on the master branch (truncated to be for authors with >2 commits, and I've merged the counts for people using different emails). Note that the counts don't reflect co-authored commits properly, and should be taken as a rough guide:

Shall we also explicitly indicate that we should try to reflect co-authored commits?

Reply:
I am glad to add something about review efforts and team programming, but I would prefer to not base this off co-authored commits. The co-authored commits method seems flawed to me because a review comment that leads to a local commit by the PR author can be equally or more helpful that suggestions that are committed on the GitHub interface and this method does not value a thorough review that does not lead to any comments.

I learned recently that you can manually add co-authors to your local commits that are recognized by GitHub by putting `Co-authored-by: Name <username@users.noreply.github.com>` but since this is not in the contributing guide not everyone is probably aware of the option, plus it is time consuming and easy to forget.

Reply:
Please see if you are OK with the changes in https://github.com/GenericMappingTools/pygmt/pull/1346/commits/ff27ac58228bb2ddbae7c197837b067f1c625e1d

Reply:
> Please see if you are OK with the changes in [ff27ac5](https://github.com/GenericMappingTools/pygmt/commit/ff27ac58228bb2ddbae7c197837b067f1c625e1d)

I didn't mean the authorship on Zenodo archives, which is okay to me.  I mean to reflect co-authors in the release note, i.e., changelog. For example, a team of four contributors contributes to a PR, but only one owns the commit. Does this case exist? If no, please forget this comment.

Reply:
> I mean to reflect co-authors in the release note, i.e., changelog.

I don't think there is an easy way to automatically list "co-authors" in the changelog.

Reply:
I tried in https://github.com/GenericMappingTools/pygmt/pull/1341#discussion_r654859661 using `git log --format="%(trailers:key=Co-authored-by)"` to get co-author names, but 1) the output is long and messy and 2) it doesn't give counts. Maybe we could find a better way?

Reply:
All the changes look good to me. We might find a better way to get co-authors' names later.